ios/engine/KMEI/KeymanEngine/Classes/KeyPreviewView.swift

```diff
@@ -0,0 +1,193 @@
+//
+//  KeyPreviewView.swift
+//  KeymanEngine
+//
+//  Created by Gabriel Wong on 2017-10-05.
+//  Copyright © 2017 SIL International. All rights reserved.
+//
+
+import UIKit
+
+public class KeyPreviewView: UIView {
+  private let keyFrame: CGRect
+  private let strokeWidth: CGFloat = 1.0
+  private let borderRadius: CGFloat = 5.0
+  private let adjX: CGFloat
+  private let adjY: CGFloat
+
+  private let borderColor = UIColor(red: 145.0 / 255.0, green: 148.0 / 255.0, blue: 152.0 / 255.0, alpha: 1.0)
+  private var bgColor = UIColor.white
+  private let bgColor2 = UIColor.white
```

Comment:
Maybe it's a Swift thing, but why is bgColor a `var` while bgColor2 is a `let`?  They're receiving the same type of value in otherwise identical manners.  I'd rather consistency here if it's reasonable.

Reply:
@jahorton in swift the the let directive is for values that will not change.  The bgColor value is changed ~ line 189.  In this case using let for values that will not change and var for those that will is the right move.  If you force them all to var, you will end up with compiler warnings in the project warning you that the value was never changed and should be defined as let.

Reply:
I think API-wise it should be a `var`. These views have code to draw a gradient in the background, but it seems like now we generally only draw a solid background. So maybe it's best to just strip the gradient code out (perhaps in a separate PR).